File: src/main/java/thedarkcolour/core/item/ItemModeled.java
package thedarkcolour.core.item;

import net.minecraft.creativetab.CreativeTabs;
import net.minecraft.item.Item;
import thedarkcolour.futuremc.FutureMC;

public class ItemModeled extends Item implements Modeled {
    public ItemModeled(String regName) {
        setTranslationKey(FutureMC.ID + "." + regName);
        setRegistryName(regName);
        addModel();
    }

    @Override
    public ItemModeled setCreativeTab(CreativeTabs tab) {
        super.setCreativeTab(tab);
        return this;
    }
}

File: src/main/java/thedarkcolour/core/item/ItemModeled.kt
package thedarkcolour.core.item

import net.minecraft.item.Item
import thedarkcolour.core.util.setItemModel
import thedarkcolour.core.util.setItemName

open class ItemModeled(regName: String) : Item() {
    init {
        setItemName(this, regName)
        setItemModel(this, 0)
    }
}